auto_dev/commands/run.py

```diff
@@ -27,62 +31,58 @@ def run(self) -> None:
         """Run the agent."""
         self.logger.info(f"Fetching agent {self.agent_name} from the local package registry...")
         self.check_tendermint()
-        self.check_agent_exists()
+        if self.check_agent_exists():
+            sys.exit(1)
         self.fetch_agent()
         self.setup_agent()
         self.execute_agent()
 
     def check_tendermint(self) -> None:
         """Check if Tendermint is running."""
+        docker_engine = docker.from_env()
+        container_name = "tm_0"
         try:
-            result = subprocess.run(
-                ["docker", "ps", "--filter", "name=tendermint", "--format", "{{.Names}}"],
-                capture_output=True,
-                text=True,
-                check=True
-            )
-            if "tendermint" not in result.stdout:
-                raise RuntimeError("Tendermint is not running.")
+            res = docker_engine.containers.get(container_name)
         except (subprocess.CalledProcessError, RuntimeError) as e:
             self.logger.error("Tendermint is not running. Please install and run Tendermint using Docker.")
             self.logger.error("You can start Tendermint with the following command:")
             self.logger.error("docker run -d --name tendermint tendermint/tendermint")
```

Comment:
We could start the tm server in the background here.
We need 2 containers the tm_init container and the tm container. 
We can leave a docker-compose in data and reference that